lib/Cake/Utility/Hash.php

```diff
@@ -870,9 +871,16 @@ public static function sort(array $data, $path, $dir = 'asc', $type = 'regular')
 
 		$dir = strtolower($dir);
 		$type = strtolower($type);
-		if ($type === 'natural' && version_compare(PHP_VERSION, '5.4.0', '<')) {
-			$type = 'regular';
+
+		// Natural and case insensitive sort is only supported from >= 5.4.0
+		if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+			if ($type === 'natural' || $type === 'natural_ignore_case' || $type === 'regular_ignore_case') {
```

Comment:
Sorry, this was my first attempt to solve it, but didn't clean it properly. Will clean it up. Thanks.